repondeur/zam_repondeur/tasks/queue.py

```diff
@@ -27,12 +27,16 @@ class State(Enum):
 
 class TransactionalHuey(Huey):
     """
-    A huey task queue that only queues tasks if the whole transaction succeeds.
+    A huey task queue that tries to play well with transactions
 
-    Also, we wrap tasks to run in their own transaction in the worker
-    (not in immediate mode).
+    - allows enqueing tasks only if the whole transaction succeeds
+    - runs each task inside an individual transaction in workers
     """
 
+    def __init__(self, *, transactional_enqueue: bool = True, **kwargs: Any) -> None:
```

Comment:
It’s True in the webapp, because we want tasks to be enqueued only when the request is a success, but it’s False in the worker, so that when the scheduler schedules a periodic task it is out in the queue immediately.

Reply:
Maybe we can improve the to make that clearer

Reply:
Maybe we can set explicitly `zam.tasks.transactional_enqueue = true` in the `[app:main]` section of ini files?

Reply:
Is my added comment making sense?